lemmy_api/src/report.rs

```diff
@@ -0,0 +1,300 @@
+use actix_web::web::Data;
+use std::str::FromStr;
+
+use lemmy_db::{comment_report::*, comment_view::*, post_report::*, post_view::*, Reportable, ReportType,};
+use lemmy_structs::{blocking, report::*};
+use lemmy_utils::{APIError, ConnectionId, LemmyError};
+use lemmy_websocket::{LemmyContext, UserOperation, messages::SendUserRoomMessage};
+
+use crate::{check_community_ban, get_user_from_jwt, is_mod_or_admin, Perform};
+
+const MAX_REPORT_LEN: usize = 1000;
+
+#[async_trait::async_trait(?Send)]
+impl Perform for CreateReport {
+  type Response = CreateReportResponse;
+
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    websocket_id: Option<ConnectionId>,
+  ) -> Result<CreateReportResponse, LemmyError> {
+    let data: &CreateReport = &self;
+    let user = get_user_from_jwt(&data.auth, context.pool()).await?;
+
+    // check size of report and check for whitespace
+    let reason = data.reason.clone();
+    if reason.trim().is_empty() {
+      return Err(APIError::err("report_reason_required").into());
+    }
+    if reason.len() > MAX_REPORT_LEN {
```

Comment:
You could probably just trim after the clone, then you wouldn't have to do `reason.trim().len()` to make sure its not a long empty reason.